components/zcash_protocol/src/constants.rs

```diff
@@ -3,3 +3,53 @@
 pub mod mainnet;
 pub mod regtest;
 pub mod testnet;
+
+// The `V<n>_TX_VERSION` constants, although trivial, serve to clarify that a
+// transaction version is meant in APIs that use a bare `u32`. Consider using
+// `zcash_primitives::transaction::TxVersion` instead.
+
+/// Transaction version 3, which was introduced by the Overwinter network upgrade
+/// and allowed until Sapling activation. It is specified in
+/// [§ 7.1 Transaction Encoding and Consensus](https://zips.z.cash/protocol/protocol.pdf#txnencoding).
+///
+/// This constant is called `OVERWINTER_TX_VERSION` in the zcashd source.
+pub const V3_TX_VERSION: u32 = 3;
+/// The version group ID for Zcash v3 transactions.
+pub const V3_VERSION_GROUP_ID: u32 = 0x03C48270;
+
+/// Transaction version 4, which was introduced by the Sapling network upgrade.
+/// It is specified in [§ 7.1 Transaction Encoding and Consensus](https://zips.z.cash/protocol/protocol.pdf#txnencoding).
+///
+/// This constant is called `SAPLING_TX_VERSION` in the zcashd source.
+pub const V4_TX_VERSION: u32 = 4;
+/// The version group ID for Zcash v4 transactions.
```

Comment:
```suggestion
/// The version group ID for Zcash v4 transactions.
///
/// This constant is called `SAPLING_VERSION_GROUP_ID` in the zcashd source.
```
Forgot to suggest this in my last review. Non-blocking.